include/particles/particle.tcc

```diff
@@ -22,7 +24,9 @@ mpm::Particle<Tdim>::Particle(Index id, const VectorDim& coord, bool status)
   this->initialise();
   cell_ = nullptr;
   nodes_.clear();
-  material_ = nullptr;
+  material_.emplace_back(nullptr);
+  material_id_.emplace_back(std::numeric_limits<unsigned>::max());
```

Comment:
Previously the vector is empty. Need to resize them, I am following the previous initialization of `material_id`.

Reply:
The `material_id` could be retrieved easily right and can be obtained from `material_`? Would it be worth it to store a vector for something we have?

Reply:
We call material_id a couple of time in particle, I think saving it may worth if we call it multiple times. Retrieving it from the material class has a lookup cost as well.

Reply:
I am OK with this then, thanks for the explanation.